3-platform/warehouse/spec/src/main/java/com/_6core/platform/warehousespec/rest/v1/InventoryManagementApi.java

```diff
@@ -0,0 +1,90 @@
+     *      - 500 (Internal Server Error) - if an unexpected error occurs during the transfer process.
+     */
+
+    @Operation(summary = "Allocate inventory", description = "Allocate a new space for the inventory in our warehouse")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "OK", content = {
+                    @Content(mediaType = "application/json",
+                            schema = @Schema(implementation = InventoryAllocateResponse.class))}),
+            @ApiResponse(responseCode = "400", description = "Field validation error"),
+            @ApiResponse(responseCode = "401", description = "Authentication error"),
+            @ApiResponse(responseCode = "404", description = "inventory not found"),
+            @ApiResponse(responseCode = "500", description = "Internal server error")
+    })
+    @PostMapping("/inventories/{inventoryId}/allocations")
+    default Mono<ResponseEntity<InventoryAllocateResponse>> inventoryAllocate (
+            String inventoryId,
+            InventoryAllocateRequest request) {
+        InventoryAllocateResponse response
+                = new InventoryAllocateResponse("-1Locate", "-1WH","-1Inv", 3);
+        return Mono.just(ResponseEntity.ok(response));
+    }
+
+    /**
+     * Supplies a product to the inventory identified by the given inventory ID.
+     *
+     * This operation allows you to add new products to an existing inventory item. The provided
+     * `inventoryId` should correspond to an existing inventory item in the system. The `products` list
+     * contains details about the products to be supplied, including their IDs and quantities.
+     *
+     *  - The response status code will be:
+     *      - 200 (OK) - if the inventory allocated is successful.
+     *      - 400 (Bad Request) - if there is a field validation error
+     *      - 401 (Unauthorized) - if there is an authentication error
+     *      - 404 (Not Found) - if the inventory item with the provided ID is not found.
+     *      - 500 (Internal Server Error) - if an unexpected error occurs during the transfer process.
+     */
+    @Operation(summary = "Supply products to inventory", description = "Supplying new products to inventory")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "OK", content = {
+                    @Content(mediaType = "application/json",
+                            schema = @Schema(implementation = InventorySupplyResponse.class))}),
+            @ApiResponse(responseCode = "400", description = "Field validation error"),
+            @ApiResponse(responseCode = "401", description = "Authentication error"),
+            @ApiResponse(responseCode = "404", description = "Inventory not found"),
+            @ApiResponse(responseCode = "500", description = "Internal server error")
+    })
+    @PatchMapping("/inventories/{inventoryId}/products")
```

Comment:
Are we changing inventory or products? I think inventory.
You can remember, that DELETE, PUT, and PATCH methods apply only to a singular resource.
That means the last element in URI is going to be `somes/{someId}`

```suggestion
    @PatchMapping("/inventories/{inventoryId}")
```